### ISSUE TITLE
Single Tower Instance vs One Per Student

### ISSUE THREAD

**Issue by mrjoshuap** [mgmt-sa-tiger-team/skylight#29]
I'd like to propose a singular Tower instance that is configured for all students, rather than multiples.  It would allow us to demonstrate multi-tenancy, teams and the like.  RBAC is a feature we might be able to easily incorporate.

Additionally, it might be less of a hit from a resource consumption on the clouds.

Thoughts?

**Comment by oatakan** [comment]
It has some advantages like you said. There are a few potential gotchas that we need to consider. One bigger subscription/license to cover all nodes across all students from a single instance. This could change based on number of students to cover. Another one is single point of failure which could be remediated with multi node deployment. Another thing to consider is that any advanced labs we may build in the future that would require admin type activities. Let's keep this issue open for a while to get more opinions on this as well.

**Comment by cigamit** [comment]
That was the original idea (not RBAC, but just create an Org per student), and it was eventually discarded for practicality purposes.  As Orcun said, being a single point of failure became an issue.  With AWS, there are multiple times were I have had VMs just stop responding and have to be rebuilt (typically I just assign the student a different student number since I build extra).  We would also have to create a larger resourced VM to handle the extra load, and licensing could be an issue.  Typically the students tower is only licensed with a "5 Node Workshop license" through the workshop form, but in order to facilitate upwards of 80+ students in a class we would need to request larger licenses.

**Comment by mrjoshuap** [comment]
Was there any discussion during the workshop this week to address this?

**Comment by cigamit** [comment]
It was discussed shortly, while I am not opposed to it being an option (which would make the playbooks a bit more complicated) it would not be the default.  There are still too many drawbacks.  We also discussed adding more VMs for the students to run against (2 -3) and this would cause licensing issues with even medium sized classes (I can't generate a license over 100 nodes myself).

We also discussed the practicality of doing even a 3 node cluster (no LB, just utilize 1 for web) to fix the single point of failure, but without fixing the licensing issue, it was a moot point.

If this is something that you want, feel free to add a pull request, and we can see how messy it gets supporting both options.
